app/Http/Controllers/DemandController.php

```diff
@@ -258,29 +258,32 @@ public function view(Request $request, string $bannerId)
         $response = new SymfonyResponse();
 
         if ($adUserEndpoint) {
-            $impressionId = $request->query->get('iid') ?: Utils::createTrackingId($this->getParameter('secret'));
 
             $demandTrackingId = Utils::attachOrProlongTrackingCookie(
                 config('app.adserver_secret'),
                 $request,
                 $response,
                 '',
-                new DateTime(),
-                $impressionId
+                new DateTime()
             );
 
             $adUserUrl = sprintf(
-                '%s/register/%s/%s/%s.gif',
+                '%s/register/%s/%s/%s.htm',
                 $adUserEndpoint,
                 urlencode(config('app.adserver_id')),
                 $demandTrackingId,
-                $impressionId
+                Utils::urlSafeBase64Encode(random_bytes(8))
             );
 
-            $response->headers->set('Location', $adUserUrl);
+        } else {
+            $adUserUrl = null;
         }
 
-        $response->send();
+        $response->setContent(view('demand/view-event', [
+            'log_url' => route('banner-context', ['id' => $eventId]),
+            'view_script_url' => url('-/view.js'),
+            'aduser_url' => $adUserUrl
+        ]));
```

Comment:
`$response->send()` was used to send response immediately (fastcgi_finish_request). Are we sure that we don't want to use it anymore? It allows to make some kind of async in php ;-)